tests/test_parser.py

```diff
@@ -30,19 +30,19 @@ def fully_parse(program: str) -> list[Structure]:
 
 def test_basic():
     assert (
-        str(fully_parse("1 1+"))
+        fully_parse("1 1+")
         == "[['none', ['number', '1']], ['none', ['number', '1']], "
         + "['none', ['general', '+']]]"
```

Comment:
is it better practice to split lines across strings using `+` or just placing them side-by-side? also, since you are not calling `str` on the result of `fully_parse`, shouldn't the right side of `==` be a list and not its string representation?

Reply:
Yes, implicit concatenation is preferred by most style guides.

```python
    assert (
        fully_parse("1 1+")
        == "[['none', ['number', '1']], ['none', ['number', '1']], " \
        "['none', ['general', '+']]]"
```